src/validation-errors/enum.js

```diff
@@ -53,18 +61,23 @@ export default class EnumValidationError extends BaseValidationError {
     const currentValue =
       dataPath === '' ? this.data : pointer.get(this.data, dataPath);
 
-    if (!currentValue) {
+    if (typeof currentValue !== 'string') {
       return null;
     }
 
-    const bestMatch = allowedValues
+    const matches = allowedValues
+      .filter(value => typeof value === 'string')
```

Comment:
`allowedValues` can be numbers too.

Reply:
Will it produce anything reasonable for numbers, though?
If we don't filter out `number` here, we will also need to adjust the condition above (line 64)